RNA-seq/06-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,430 @@
+}
+```
+
+#### Input files
+
+```{r input_files}
+# The metadata describing the samples
+histologies_file <- file.path(data_dir, "pbta-histologies-subset.tsv")
+
+# The RNAseq counts table
+rnaseq_file = file.path(data_dir, "pbta-rsem-expected_count-subset.rds")
+```
+
+#### Output files
+
+```{r png_output, live = TRUE}
+heatmap_file <- file.path(plots_dir, 
+                          "common_histologies_high_variance_heatmap.png")
+```
+
+## Read in and prepare data
+
+### Metadata 
+
+Let's read in the metadata file file and take a look at the data.
+
+```{r read_rds, live = TRUE}
+histologies_df <- read_tsv(histologies_file)  
+```
+
+Use the chunk below to explore the metadata data frame.
+
+```{r explore_histologies_df, live = TRUE, eval = FALSE}
+View(histologies_df)
+```
+
+We'll use the disease labels in a column called `short_histology` when we label the heatmap.
+Let's count how many samples are assigned each `short_histology` label using the Tidyverse.
+
+```{r count_histologies}
+histology_count_df <- histologies_df %>%
+  # Count how many samples are in each short_histology and name the column 
+  # with that number n
+  count(short_histology) %>%
+  # Sort from largest number of samples to smallest number of samples in a 
+  # histology
+  arrange(desc(n))
+
+histology_count_df
+```
+
+### RNA-seq data
+
+Read in the expression count matrix (stored as a data frame).
+
+```{r read_in_rnaseq, live = TRUE}
+# Read in and examine the RNA-seq data
+rnaseq_exp <- read_rds(rnaseq_file)
+```
+
+### Convert and round
+
+The count data we have is mostly integers, but because of the estimation procedure that RSEM uses, some is not. 
+`DESeq2`, which we will be using below expects all integers, so we will round all of the count data. 
+This is easiest if we first convert from a data frame to a matrix.
+
+```{r convert_round, live = TRUE}
+rnaseq_mat <- rnaseq_exp %>%
+  # move gene_id to the rownames
+  tibble::column_to_rownames("gene_id") %>%
+  # convert to a matrix and round
+  as.matrix() %>%
+  round()
+```
+
+### Variance Stabilizing Transformation
+
+Raw count data is not usually suitable for the algorithms we use for clustering and heatmap display. 
+To improve this, we will transform the count data to create an expression measure that is better suited for these analyses. 
+The core transformation will map the expression to a log2 scale, while accounting for some of the expected variation among samples and genes.
+
+Since different samples are usually sequenced to different depths, we want to transform our RNA-seq count data to make different samples more directly comparable. 
+We also want to deal with the fact that genes with low counts are also likely to have higher variance (on the log2 scale), as that could bias our clustering.
+To handle both of these considerations, we can calculate a Variance Stabilizing Transformation of the count data, and work with that transformed data for our analysis.
```

Comment:
This is not the first time they will see this transformation, should this explanation be moved here instead: https://github.com/AlexsLemonade/training-modules/blob/b989631516d393f1707d321107025ec25afc6ee4/RNA-seq/03-gastric_cancer_exploratory.Rmd#L89

We also cover VST in the first RNA-seq exercise notebook. The point there is to visualize what the transformation does to the underlying values! Starts here: https://github.com/AlexsLemonade/training-modules/blob/b989631516d393f1707d321107025ec25afc6ee4/RNA-seq/exercise_01-exploratory_data_analysis.Rmd#L126

Reply:
Converting to a `DESeqDataSet` from a matrix here makes sense to me! Because I know that you have another branch with PCA for this dataset, I will go ahead and suggest that should you have put a more detailed description of PCA in that notebook, you can move it to `03` also? I do still think there may be value in comparing and contrasting hierarchical clustering with PCA, time permitting.